cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/template/etl/api/TransformContext.java

```diff
@@ -41,4 +41,41 @@
    * @return {@link Metrics} for collecting metrics
    */
   Metrics getMetrics();
+
+  /**
+   * Creates a new instance of a plugin.
+   * The instance returned will have the {@link co.cask.cdap.api.templates.plugins.PluginConfig} setup with
+   * {@link PluginProperties} provided at the time when the
+   * {@link co.cask.cdap.api.templates.AdapterConfigurer#usePlugin(String, String, String, PluginProperties)}
+   * was called during the
+   * {@link co.cask.cdap.api.templates.ApplicationTemplate#configureAdapter(String,
+   * Object, co.cask.cdap.api.templates.AdapterConfigurer)} time.
```

Comment:
Looks like this one is different from the next because it is called at the time of AdapterConfiguration rather than program configuration time, right? You can improve this by adding a label in the Javadoc comment to the link so that the link goes to the same place but with better text
```
{@link co.cask.cdap.api.templates.ApplicationTemplate#configureAdapter(String,
Object, co.cask.cdap.api.templates.AdapterConfigurer) adapter configuration} time.